transport_analysis/tests/test_viscosity.py

```diff
@@ -0,0 +1,196 @@
+import pytest
+from numpy.testing import assert_allclose
+
+from transport_analysis.viscosity import (
+    ViscosityHelfand as VH,
+)
+import MDAnalysis as mda
+from MDAnalysis.units import constants
+from MDAnalysis.transformations import set_dimensions
+import numpy as np
+
+from MDAnalysis.exceptions import NoDataError
+from MDAnalysisTests.datafiles import PRM_NCBOX, TRJ_NCBOX, PSF, DCD
+
+
+@pytest.fixture(scope="module")
+def u():
+    return mda.Universe(PRM_NCBOX, TRJ_NCBOX)
+
+
+@pytest.fixture(scope="module")
+def ag(u):
+    return u.select_atoms("name O and resname WAT and resid 1-10")
+
+
+@pytest.fixture(scope="module")
+def u_no_vels():
+    return mda.Universe(PSF, DCD)
+
+
+@pytest.fixture(scope="module")
+def ag_no_vels(u_no_vels):
+    return u_no_vels.select_atoms("backbone and name CA and resid 1-10")
+
+
+@pytest.fixture(scope="module")
+def NSTEP():
+    nstep = 5001
+    return nstep
+
+
+@pytest.fixture(scope="module")
+def visc_helfand(ag):
+    vh_t = VH(ag, fft=False)
+    vh_t.run()
+    return vh_t
+
+
+# Full step trajectory of unit velocities i.e. v = 0 at t = 0,
+# v = 1 at t = 1, v = 2 at t = 2, etc. for all components x, y, z
+# with mass, positions, and volume
+@pytest.fixture(scope="module")
+def step_vtraj_full(NSTEP):
+    # Set up unit velocities
+    v = np.arange(NSTEP)
+    velocities = np.vstack([v, v, v]).T
+    # NSTEP frames x 1 atom x 3 dimensions, where NSTEP = 5001
+    velocities_reshape = velocities.reshape([NSTEP, 1, 3])
+
+    # Positions derived from unit velocity setup
+    x = np.arange(NSTEP).astype(np.float64)
+    # Since initial position and velocity are 0 and acceleration is 1,
+    # position = 1/2t^2
+    x *= x / 2
+    positions = np.vstack([x, x, x]).T
+    # NSTEP frames x 1 atom x 3 dimensions, where NSTEP = 5001
+    positions_reshape = positions.reshape([NSTEP, 1, 3])
+    u = mda.Universe.empty(1, n_frames=NSTEP, velocities=True)
+
+    # volume of 8.0
+    dim = [2, 2, 2, 90, 90, 90]
+    for i, ts in enumerate(u.trajectory):
+        u.atoms.velocities = velocities_reshape[i]
+        u.atoms.positions = positions_reshape[i]
+        set_dimensions(dim)(u.trajectory.ts)
+
+    # mass of 16.0
+    u.add_TopologyAttr("masses", [16.0])
+    return u
+
+
+def characteristic_poly_helfand(
+    test_universe,
+    stop,
+    n_dim,
+    temp_avg=300.0,
+    mass=16.0,
+    vol_avg=8.0,
+    start=0,
+    step=1,
+):
+    # update when mda 2.6.0 releases with typo fix
+    try:
```

Comment:
Grab the actual value here.

Reply:
Was wondering also - why is grabbing the actual value better here?

Reply:
Just to reduce code volume, you still have to index the dict later with this approach.